go/store/types/linestring.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/dolthub/dolt/go/store/hash"
+)
+
+const (
+	LengthSize = 4
+)
+
+// Linestring is a Noms Value wrapper around a string.
+type Linestring struct {
+	SRID   uint32
+	Points []Point
+}
+
+// Value interface
+func (v Linestring) Value(ctx context.Context) (Value, error) {
+	return v, nil
+}
+
+func (v Linestring) Equals(other Value) bool {
+	// Compare types
+	v2, ok := other.(Linestring)
+	if !ok {
+		return false
+	}
+	// Compare SRID
+	if v.SRID != v2.SRID {
+		return false
+	}
+	// Compare lengths of points
+	if len(v.Points) != len(v2.Points) {
+		return false
+	}
+	// Compare each point
+	for i := 0; i < len(v.Points); i++ {
+		if !v.Points[i].Equals(v2.Points[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+func (v Linestring) Less(nbf *NomsBinFormat, other LesserValuable) (bool, error) {
+	// Compare types
+	v2, ok := other.(Linestring)
+	if !ok {
+		return LinestringKind < other.Kind(), nil
+	}
+	// TODO: should I even take this into account?
+	// Compare SRID
+	if v.SRID != v2.SRID {
```

Comment:
Regarding the TODO, I'd make linestrings in MySQL with their only difference being their SRID. Insert them in some random order, then query with an `ORDER BY` and see if MySQL takes it into account. If it does order them, then do another one but with the SRID in descending order and the linestring values in ascending order (or vice versa, as long as they're opposite). Then do another `ORDER BY` and it will tell you the priority (whether we should compare points first or the SRID first), as right now you have the SRID first (which may be correct).

Test this (and the other ones if needed, don't see a TODO on those) and make a new PR with the changes (if any). We 100% need to know before we remove the feature flag though.